extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -685,8 +685,15 @@ public static Table readPartitionedTableInferSchema(
                 throw new IllegalArgumentException("First location key " + firstKey
                         + " has null partition value at partition key " + partitionKey);
             }
-            allColumns.add(ColumnDefinition.fromGenericType(partitionKey,
-                    getUnboxedTypeIfBoxed(partitionValue.getClass()), null, ColumnDefinition.ColumnType.Partitioning));
+
+            // Primitives should be unboxed, except booleans
```

Comment:
Why?

Reply:
I believe this is because other types have a null sentinel, but in order for booleans to have a possible null, we need to box them instead (it would be silly to have false mean "not set", etc). This is a convention followed in other parts of the platform, and the check was just missing here (but is present in another path to creating parquet tables, linked in the issue).